Documentation/General/productVision.tex

```diff
@@ -0,0 +1,171 @@
+
+\item The backend keeps track of which rooms are active and what song is currently played in the room.
+Furthermore, the backend is responsible for transmitting the chat messages between the users in the room.
+
+\item The music matching algorithm is used to compare and match moods of songs and chatrooms.
+It has two roles: list the room suggestions for a user and provide the next song to be played for a room.
+\end{enumerate}
+
+\newpage
+
+\subsection{Frontend}
+
+The MoodCat frontend will be web-based.
+This allows us to support all modern devices which enables a large target audience.
+We will develop the frontend using modern techniques as HTML5\cite{HTML}, CSS3\cite{CSS} and AngularJS\cite{AngularJS}.
+Users can login using their Soundcloud account.
+This allows us to integrate with the Soundcloud platform and prevents having to register for our services first.
+
+\subsection{Backend}
+
+The MoodCat backend will be a RESTful API written in Java using JAX-RS. It will be backed by a relational (PostgreSQL) database. The backend connects the frontend to the music matching algorithm.
+
+\subsection{Music matching algorithm}
+
+The Music matching algorithm determines and compares moods for songs, rooms and users.
+Moodcat will be developed around a learning algorithm.
+This means that Moodcat will improve over time.
+Default values will be determined by examining the audio features provided by our data set.
+Then, through a system of user votes, the system will learn and improve over time.
+
+We discovered that moods and emotions can be mapped to a valence/arousal space (see figure \ref{fig:avgraph}).
+This graph appears to be used in a lot of scientifical papers \cite{Kosina} \cite{McVicar}.
+
+\begin{figure}[h]
+\includegraphics[scale=0.75]{avgraph.jpg}
+\caption{The valance/arousal graph \cite{Book}}
+\label{fig:avgraph}
+
+\end{figure}
+
+Arousal and valence are physiological terms.
+Arousal means how calm or excited you are.
+Valence is the amount of attraction you have towards a certain event.
+
+In order to use arousal and valence, we have to express them in terms of music features.
+For valence we can use harmony \cite{PresentationMER},mode  \cite{PresentationMER} and tonality  \cite{PresentationMER}.
+For arousal we can use pitch \cite{PresentationMER} ,timbre \cite{PresentationMER}, tempo \cite{PresentationMER} and loudness \cite{PaperME}.
+
+\newpage
+
+The data of EEMCS does not explicitly provide the above features. However, we can extract some of them from the low-level features the data does provide.
+
+The low-level features we can use for valence:
+\begin{enumerate}
+\item For harmony we can use the dissonance of a song.
+\item To determine the mode of a song, we can use the key.
+\item Finally the tonal provides the information to obtain the tonality.
+\end{enumerate}
+The low-level features we can use for arousal:
+\begin{enumerate}
+\item BMP gives us insight of the tempo of a song
```

Comment:
This is correct but confusing on first read. Why is BMP not equal to the tempo (I would use speed here by the way)? This is because not only BMP matters but also the time signature (Dutch: maatsoort) and rhythm influences how we experience the liveliness of music.